xmtp_id/Cargo.toml

```
@@ -18,6 +18,7 @@ ed25519-dalek = { workspace = true, features = ["digest"] }
 ethers = { workspace = true, features = ["rustls"] }
 futures.workspace = true
 hex.workspace = true
+lru = "0.13.0"
```

Let's make this optional, and add an "lru" feature down below in this file that the validation service can turn on.
```suggestion
lru = { version = "0.13.0", optional = true }
```

---

And down below under features, add
```suggestion
scw_cache = ["lru"]
```

---

On the fence about it though. Curious what others think

---

if this is only used in `mls_validation_service`, i think the rustier way to do this would be to prefer [composition](https://www.hackingwithrust.net/2023/10/23/a-composite-pattern-in-rust/)

Something like:

- leave xmtp_id alone
- create a new mod in mls_validation_service `scw_cache` or something
- create a new struct to contain the MultiSmartContract Verifier + the cache:
- `struct CachedScwVerifier { inner: MultiSmartContractSignatureVerifier, cache: /* */}`
- impl the caching on `CachedScwVerifier` by implementing the trait `SmartContractSignatureVerifier`on `CachedScwVerifier`

This has the nice property of being composable. We can now choose whether we want a cached version of "MultiSmartContractVerifier" or a non-cached version. Further, if we decide that there are other consumers of the cached version, we can easily copy paste  it into a more core crate like xmtp_id.